src/catalogs/synchronous/list_items.ts

```diff
@@ -1,24 +1,59 @@
-import { get } from '../../common/request'
-import { buildParams } from '../../common/request/params'
+import fetch, { Response } from 'node-fetch'
+
+import { ResponseError } from '../../common/request'
 import { CatalogListItem, CatalogListItemsBody, CatalogListItemsResponse } from './types'
 
+function getNextPageLink(linkHeader: string | null): string | undefined {
+  const linkMatches = linkHeader?.matchAll(/<([^>]+)>; rel="(prev|next)"/g)
+  if (linkMatches) {
+    for (const match of linkMatches) {
+      if (match[2] === 'next') {
+        return match[1]
+      }
+    }
+  }
+}
+
 /**
  * Request catalog items.
  *
  * {@link https://www.braze.com/docs/api/endpoints/catalogs/catalog_items/synchronous/get_catalog_items_details_bulk/}
  */
-export function listCatalogItems<T extends CatalogListItem>(
+export async function listCatalogItems<T extends CatalogListItem>(
   apiUrl: string,
   apiKey: string,
-  body: CatalogListItemsBody,
+  { catalog_name, max_pages }: CatalogListItemsBody,
```

Comment:
One question here. I almost want to put in a safe guard here to maybe default max_pages to 100 as I don't think anyone would reasonably use this endpoint if there are that many pages. I could even see 50 or less. Do you care?

```suggestion
  { catalog_name, max_pages = 100 }: CatalogListItemsBody,
```

Reply:
So my question is what happens without the safeguard? Do we hit a rate limit? If it's not likely that this can be misused, then I'd vote for keeping it as-is

Reply:
There's a rate limit of 50 requests per minute.

https://www.braze.com/docs/api/endpoints/catalogs/catalog_items/synchronous/get_catalog_items_details_bulk

I'm good at leaving it as is and put the onus on the dev to know what they are doing.

Thanks.

Reply:
Untested, just threw this together, but something like this might be more robust.

```typescript
class CatalogListItems<T extends CatalogListItem> {
  constructor(
    private readonly apiUrl: string,
    private readonly apiKey: string,
    public readonly items: T,
    private readonly nextPageLink?: string,
  ) {}

  hasNextPage(): boolean {
    return !!this.nextPageLink
  }

  async next(): Promise<CatalogListItems<T>> {
    if (!this.nextPageLink) {
      throw new ResponseError('There is no next page', 0)
    }
    return CatalogListItems.getItems(this.apiUrl, this.apiKey, this.nextPageLink)
  }

  static async getItems<T extends CatalogListItem>(
    apiUrl: string,
    apiKey: string,
    url: string,
  ): Promise<CatalogListItems<T>> {
    const response: Response = await fetch(apiUrl + url, {
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${apiKey}`,
      },
    })
    const data = await response.json()

    if (!response.ok) {
      throw new ResponseError(data.message, response.status, data.errors)
    }

    return new CatalogListItems<T>(
      apiUrl,
      apiKey,
      data.items,
      getNextPageLink(response.headers.get('Link')),
    )
  }
}

/**
 * Request catalog items.
 *
 * {@link https://www.braze.com/docs/api/endpoints/catalogs/catalog_items/synchronous/get_catalog_items_details_bulk/}
 */
export function getListCatalogItems<T extends CatalogListItem>(
  apiUrl: string,
  apiKey: string,
  { catalog_name }: CatalogListItemsBody,
): Promise<CatalogListItems<T>> {
  return CatalogListItems.getItems<T>(apiUrl, apiKey, `/catalogs/${catalog_name}/items`)
}
```

I'll test it out when I'm online later today.

Reply:
Sounds like a plan